test/distrib/cpp/run_distrib_test_raspberry_pi.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+cd "$(dirname "$0")/../../.."
+grpc_dir=$(pwd)
+
+echo "deb http://archive.debian.org/debian jessie-backports main" | tee /etc/apt/sources.list.d/jessie-backports.list
+echo 'Acquire::Check-Valid-Until "false";' > /etc/apt/apt.conf
+sed -i '/deb http:\/\/deb.debian.org\/debian jessie-updates main/d' /etc/apt/sources.list
+apt-get update
+apt-get install -t jessie-backports -y git libssl-dev wget
+
+# Install CMake 3.16
+wget -q -O cmake-linux.sh https://github.com/Kitware/CMake/releases/download/v3.16.1/cmake-3.16.1-Linux-x86_64.sh
+sh cmake-linux.sh -- --skip-license --prefix=/usr
+rm cmake-linux.sh
+
+# Build and install gRPC for the host architecture.
+# We do this because we need to be able to run protoc and grpc_cpp_plugin
+# while cross-compiling.
+mkdir -p "cmake/build"
+pushd "cmake/build"
+cmake \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DgRPC_INSTALL=ON \
+  -DgRPC_BUILD_TESTS=OFF \
+  -DgRPC_SSL_PROVIDER=package \
+  ../..
+make -j4 install
+popd
+
+# Download raspberry pi toolchain.
+mkdir -p "/tmp/raspberrypi_root"
+pushd "/tmp/raspberrypi_root"
+git clone https://github.com/raspberrypi/tools raspberrypi-tools
+cd raspberrypi-tools && git checkout 4a335520900ce55e251ac4f420f52bf0b2ab6b1f && cd ..
+
+# Write a toolchain file to use for cross-compiling.
+cat > toolchain.cmake <<'EOT'
+SET(CMAKE_SYSTEM_NAME Linux)
```

Comment:
qq: don't the raspberry pi tools provide  a ready-to-use toolchain?

Reply:
They do, that's what we're cloning from github in this distrib test.

OTOH, CMake toolchain files are really meant to be machine-specific. As you can see in this example, they tend to have hardcoded paths to compilers & such in them.